test_util/azure_test_driver.py

```diff
@@ -124,14 +120,17 @@ def main():
         print("Creating template deployment ...")
         deploy_poller = rmc.deployments.create_or_update(group_name, deployment_name, deployment_properties)
 
-        # Stop after 45 attempts (each one takes up to one minute)
-        @retry(stop_max_attempt_number=45)
+        def retry_if_assertion_error(exception):
+            """Return True if we should retry (in this case when it's an AssertionError), False otherwise"""
+            return isinstance(exception, AssertionError)
+
+        @retry(retry_on_exception=retry_if_assertion_error, stop_max_attempt_number=45)
```

Comment:
The retry interval is defined by the thread wait time and assertion in the below.

```
             res = deploy_poller.result(timeout=60)
             ...
             assert deploy_poller.done(), "Not done deploying."		    
```

So, after every 60 seconds and assertion is made if the task is done and this is done for 45 times.

* Previously, the 45 times retry was done for any failure.
* Now, this retry of task is done only only carried out for `AssertionError`, any other Infrastructure issues like `CloudError` will fail fast.

Hope this helps.